modules/cloudsql-instance/main.tf

```diff
@@ -52,7 +52,7 @@ resource "google_sql_database_instance" "primary" {
   region              = var.region
   database_version    = var.database_version
   encryption_key_name = var.encryption_key_name
-  root_password       = var.root_password
+  root_password       = var.root_password.random_password ? random_password.root_password[0].result : var.root_password.password
```

Comment:
This will fail if `var.root_password` is null. Make the variable `nullable = false` to avoid having to check for null values.